yunwoo/programmers-1/이상한-문자-만들기.js

```diff
@@ -0,0 +1,48 @@
+function solution(s) {
+  const strArr = s.split(" ");
+  let result = "";
+
+  for (let i = 0; i < strArr.length; i++) {
+    for (let j = 0; j < strArr[i].length; j++) {
+      result += !(j % 2)
+        ? strArr[i][j].toUpperCase()
+        : strArr[i][j].toLowerCase();
+    }
+
+    if (i < strArr.length - 1) {
+      result += " ";
+    }
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(s) {
  const words = s.toLowerCase().split(" ");
  const result = [];

  for (let i = 0; i < words.length; i++) {
    let temp = "";

    for (let j = 0; j < words[i].length; j++) {
      temp += !(j % 2) ? words[i][j].toUpperCase() : words[i][j];
    }
    result[i] = temp;
  }

  return result.join(" ");
}
```
- 풀이 잘봤습니다 :) 우선 strArr보다는 단어들을 여러 개 배열로 모았다는 의미로 `words`가 어떨까요! 🙆🏻‍♂️ 그리고 반복문에서 toLowerCase()말고 애초에 문자열 s를 split전에 toLowerCase를 해서 반복문에서 코드양을 최적화해봤습니다 🙇🏻‍♂️
- 결과를 도출 할 때 if문을 통한 활용도 좋았지만, 공백을 배열 + join 메서드를 활용하는 방법도 존재합니다 위에 풀이는 temp와 result를 배열로 이용한 풀이입니다.
- 아래 풀이는 map을 이용한 풀이입니다 참고만 해주세요
```js
function solution(s) {
  return s
    .toLowerCase()
    .split(" ")
    .map((word) => {
      return word
        .split("")
        .map((el, i) => (i % 2 === 0 ? el.toUpperCase() : el))
        .join("");
    })
    .join(" ");
}
```

Reply:
- words !! 훨씬 좋은거 같습니다👍
- 아 애초에 소문자로 만들어주었으면 되는군요.. 히든케이스에 쫓기다보니 생각 못한거 같습니다ㅎㅎ..
- 와 `join`메서드에 공백을 넣어주면 각 요소 사이에 공백이 생기는군요!? 처음알았습니다 와 감사합니다🙇‍♂️🙇‍♂️
- 역시 다중`map`으로도 가능하군요.. 좋은 참고가 됐습니다 리뷰 감사합니다!!!!